brave-http-tests/src/main/java/brave/http/StrictCurrentTraceContext.java

```diff
@@ -0,0 +1,27 @@
+package brave.http;
+
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class StrictCurrentTraceContext implements CurrentTraceContext {
```

Comment:
Added this as the default implementation being an inheritable thread local makes it hard to test if propagation works by accident (ex only because inheritable thread local is used), or if because it was coded correctly. You'll notice this also is defensive about being closed on the same thread as it was opened.